File: problems/p9/p9.py
"""
project euler problem 9
"""
def is_triplet(a, b, c):
    if a*a + b*b == c*c:
        return True
    return False

def equals_a_num(a, b, c, num):
    if a + b + c == num:
        return True
    return False

def trip(num):
    for c in range(1, num//2):
        for b in range(1, c):
            for a in range(1, b):
                if is_triplet(a, b, c) and equals_a_num(a, b, c, num):
                    print((a, b, c))
                    return a*b*c

print(trip(1000))

File: problems/p9/p9.cpp
/*
project euler problem 9
*/
#include <iostream>

bool isTriplet(int a, int b, int c){
  if (a*a + b*b == c*c) {
    return true;
  }
  return false;
}

bool equalsANum(int a, int b, int c, int num){
  if (a + b + c == num) {
    return true;
  }
  return false;
}

int main(int argc, char const *argv[]) {
  int num = 1000;
  for (size_t c = 1; c < num/2; c++) {
    for (size_t b = 1; b < c; b++) {
      for (size_t a = 1; a < b; a++) {
        if (isTriplet(a, b, c) && equalsANum(a, b, c, num)) {
          std::cout << "(" << a << "," << b << "," << c << ")" << '\n';
          std::cout << a*b*c << '\n';
        }
      }
    }
  }
  return 0;
}
